File: LeetCode/go/LC3402_MinimumOperationsToMakeColumnsStrictlyIncreasing/minimum_operations_to_make_columns_strictly_increasing.go
package minimum_operations_to_make_columns_strictly_increasing

func minimumOperations(grid [][]int) int {
	counter := 0
	for i := 0; i < len(grid[0]); i++ {
		for j := 0; j < len(grid)-1; j++ {
			delta := grid[j+1][i] - grid[j][i]
			if delta <= 0 {
				grid[j][i] += -delta + 1
				counter += -delta + 1
			}
		}
	}
	return counter
}

File: LeetCode/java/LC3402_Minimum_Operations_to_Make_Columns_Strictly_Increasing/MinimumOperationsToMakeColumnsStrictlyIncreasing.java
package LC3402_Minimum_Operations_to_Make_Columns_Strictly_Increasing;

/**
 * <a href="https://leetcode.cn/problems/minimum-operations-to-make-columns-strictly-increasing/">Minimum Operations to Make Columns Strictly Increasing</a>
 */
public class MinimumOperationsToMakeColumnsStrictlyIncreasing {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.minimumOperations(new int[][]{{1, 2}, {3, 4}}));
    }
}

class Solution {
    public int minimumOperations(int[][] grid) {
        int counter = 0;
        for (int i = 0; i < grid[0].length; i++) {
            for (int j = 1; j < grid.length; j++) {
                int delta = grid[j][i] - grid[j - 1][i];
                if (delta <= 0) {
                    grid[j][i] += -delta + 1;
                    counter += -delta + 1;
                }
            }
        }
        return counter;
    }
}